airflow/providers/google/cloud/operators/vertex_ai/supervised_fine_tuning.py

```diff
@@ -0,0 +1,116 @@
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields = ("location", "project_id", "impersonation_chain", "train_dataset", "validation_dataset")
+
+    def __init__(
+        self,
+        *,
+        source_model: str,
+        train_dataset: str,
+        project_id: str,
+        location: str,
+        tuned_model_display_name: str | None = None,
+        validation_dataset: str | None = None,
+        epochs: int | None = None,
+        adapter_size: int | None = None,
+        learning_rate_multiplier: float | None = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.source_model = source_model
+        self.train_dataset = train_dataset
+        self.tuned_model_display_name = tuned_model_display_name
+        self.validation_dataset = validation_dataset
+        self.epochs = epochs
+        self.adapter_size = adapter_size
+        self.learning_rate_multiplier = learning_rate_multiplier
+        self.project_id = project_id
+        self.location = location
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context: Context):
+        self.hook = SupervisedFineTuningHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+        )
+        response = self.hook.train(
+            source_model=self.source_model,
+            train_dataset=self.train_dataset,
+            project_id=self.project_id,
+            location=self.location,
+            validation_dataset=self.validation_dataset,
+            epochs=self.epochs,
+            adapter_size=self.adapter_size,
+            learning_rate_multiplier=self.learning_rate_multiplier,
+            tuned_model_display_name=self.tuned_model_display_name,
+        )
+
+        self.log.info("Tuned Model Name: %s", response.tuned_model_name)
+        self.log.info("Tuned Model Endpoint Name: %s", response.tuned_model_endpoint_name)
+
+        self.xcom_push(context, key="tuned_model_name", value=response.tuned_model_name)
+        self.xcom_push(context, key="tuned_model_endpoint_name", value=response.tuned_model_endpoint_name)
```

Comment:
Could you add a return result? I think in your case it should be `SupervisedTuningJob` in dict format. For example how it looks for Pipeline Job: https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/operators/vertex_ai/pipeline_job.py#L204C1-L206C1

Reply:
Thanks for the advice, added in latest commits